jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java

```diff
@@ -14,4 +20,57 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public int update(final String sql, final Object... args) {
+        try (final Connection conn = dataSource.getConnection();
+             final PreparedStatement pstmt = prepareMappedStatement(conn, sql, args);) {
+
+            return pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.warn("쿼리 실행 도중에 오류가 발생하였습니다.", e);
```

Comment:
로그 메세지에 대해 궁금한 점이 있는데요. 해당 부분은 insert 쿼리 부분이라 쿼리 실행 도중에 오류가 발생했다는 메세지를 남기는 것이고, 아래의 메서드들은 select 쿼리라 조회 도중 에러가 발생했다고 로그를 남기는 것일까요?

제 생각에는 구분할 것 없이 공통된 로깅 메세지를 남기고, 추가로 어떤 sql인지 로깅을 남기는 것이 덜 귀찮을 것 같다는 생각인데요. 헤나는 어떻게 생각하시나요?
그냥 제가 귀찮은게 싫어서 그런 것일수도 있습니다.
```suggestion
            log.warn("쿼리 실행 도중에 오류가 발생하였습니다. sql : {}", sql, e);
```

Reply:
말씀해주신 것처럼
update 메서드는 CUD에 해당해서 '쿼리 실행 도중' 이라는 공통된 로그 메시지로 만들었고
select 같은 경우는 '조회 도중'이라고만 로그 메시지를 남겼습니다.

사실 자세한 로그 메시지를 남기지 않고 있어서 insert, select 모두 공통된 로그 메시지여도 상관없다고 생각됩니다!

sql 로깅 같은 경우는 따로 작성하지 않아도 예외랑 같이 나와서 생략했었습니다.
```sql
-- 잘못된 sql
selectds id, 
```
![image](https://github.com/woowacourse/jwp-dashboard-jdbc/assets/82203978/cb46d842-b6c7-4ec2-9de6-a25a7eb52079)

근데 박스터 의견을 듣고나니까 로깅을 좀 더 자세히 해놓자는 생각이 들게 됐습니다. 👍👍

그래서 아래와 같이 sql과 들어가는 값까지 로깅하는 방식으로 수정했습니다.

```java
// SQL에 변수를 매핑하는 경우
// ...
        } catch (SQLException e) {
            log.warn("쿼리 실행 도중에 오류가 발생하였습니다. {} ====> SQL = {} {} ====> Args = {}",
                    System.lineSeparator(), sql, System.lineSeparator(), args, e);
            throw new RuntimeException(e);
        }
// ...

// SQL에 변수를 매핑하지 않는 경우
// ...
        } catch (SQLException e) {
            log.warn("조회하던 도중에 오류가 발생하였습니다. {} ====> SQL = {}",
                    System.lineSeparator(), sql, e);
            throw new RuntimeException(e);
        }
// ...
```

아래와 같이 로깅 나오게 수정했습니다.

```text
13:32:48.413 [WARN ] [Test worker] [o.s.jdbc.core.JdbcTemplate] - 쿼리 실행 도중에 오류가 발생하였습니다. 
 ====> SQL = selectds id, account, password, email from users where id = ? 
 ====> Args = [1]
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "[*]selectds id, account, password, email from users where id = ?"; expected "SAVEPOINT, SCRIPT, SHUTDOWN, SHOW"; SQL statement:
selectds id, account, password, email from users where id = ? [42001-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
```